src/frontend/components/StudentLiveControlButtons/StudentLeaveDiscussionButton/index.spec.tsx

```diff
@@ -0,0 +1,64 @@
+import { fireEvent, render, screen } from '@testing-library/react';
+import React from 'react';
+
+import { PLAYER_ROUTE } from 'components/routes';
+import { useParticipantWorkflow } from 'data/stores/useParticipantWorkflow';
+import { wrapInIntlProvider } from 'utils/tests/intl';
+import { wrapInRouter } from 'utils/tests/router';
+import * as mockWindow from 'utils/window';
+
+import { StudentLeaveDiscussionButton } from '.';
+
+jest.mock('utils/window', () => ({
+  converse: {
+    participantLeaves: jest.fn(),
+  },
+}));
+jest.mock('components/SVGIcons/JoinDiscussionSVG', () => ({
+  JoinDiscussionSVG: () => <span>join discussion icon</span>,
+}));
+
+describe('<StudentLeaveDiscussionButton />', () => {
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('renders the leave button', () => {
+    render(wrapInIntlProvider(wrapInRouter(<StudentLeaveDiscussionButton />)));
+
+    screen.getByRole('button', { name: 'Leave discussion' });
+    screen.getByText('join discussion icon');
+  });
+
+  it('clicks on the leave button and is redirected to the player', () => {
+    const mockRest = jest.fn();
+    useParticipantWorkflow.setState({
+      reset: mockRest,
+    });
+
+    render(
+      wrapInIntlProvider(
+        wrapInRouter(<StudentLeaveDiscussionButton />, [
+          {
+            path: PLAYER_ROUTE(),
+            render: () => {
+              return <span>{'video player'}</span>;
+            },
+          },
+        ]),
+      ),
+    );
+
+    const leaveButton = screen.getByRole('button', {
+      name: 'Leave discussion',
+    });
+    screen.getByText('join discussion icon');
```

Comment:
Same here. You don't join the discussion

Reply:
https://github.com/openfun/marsha/pull/1249/commits/3913518d96f998067e5379d26f9dddc9a2f6a5c0